lint: Implement HTML Linting with htmlhint and fix indent.

vsvipul opened the pull request:
**What's this PR do?**
Implements HTML linting using htmlhint and uses
indentation rules of zulip webapp - 4 spaces.
Creates a separate file .htmlhintrc for the rules, most
of which are defaults. Also, fixes indentation in html
files and adds a missing title in about.html.

Fixes #676.

All rules for htmlhint are listed [here](https://github.com/htmlhint/HTMLHint/wiki/Rules).

**You have tested this PR on:**
  - [x] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

vsvipul committed:
lint: Implement HTML Linting with htmlhint and fix indent.

Implements HTML linting using htmlhint and uses
indentation rules of zulip webapp - 4 spaces.
Creates a separate file .htmlhintrc for the rules, most
of which are defaults. Also, fixes indentation in html
files and adds a missing title in about.html.

Fixes #676.

kanishk98 commented on package.json:
```diff
@@ -130,6 +131,7 @@
     "google-translate-api": "2.3.0",
     "gulp": "^4.0.0",
     "gulp-tape": "0.0.9",
+    "htmlhint": "^0.11.0",
```
I've been asked to include the exact version number a couple of times when adding a module - I'd check with the maintainers once. 
LGTM otherwise.

vsvipul replied:
Used same for stylelint previously. Will have to change that too if this is the case.

akashnimare replied:
Yeah, we prefer to use the exact working version at least for app's dependencies. @vsvipul can you change this and also change other dependencies version (gulp etc).

vsvipul replied:
Sure, will do..

vsvipul replied:
@akashnimare changed to follow exact versioning for all modules.

vsvipul committed:
deps: Change versioning of various dependencies to exact versions.

As we prefer to use exact working version
for dependencies, changed the versions to
exact versions.

akashnimare commented:
Looks good. @kanishk98 can you test this on Windows?

kanishk98 commented:
>  @kanishk98 can you test this on Windows?

On it, working with Vipul to debug some issues.

vsvipul commented:
@akashnimare as told by @kanishk98 , on windows we need to change 
```
    "lint-html": "./node_modules/.bin/htmlhint 'app/renderer/*.html' ",
```
to
```
    "lint-html": "./node_modules/.bin/htmlhint app/renderer/*.html ",
```
to make it run properly.

On running the second command, i get multiple config loaded files on Ubuntu, while on Windows, we only get a single line.

Output for Ubuntu-
```
> zulip@2.5.0-beta lint-html /home/vipul/Documents/zulip-electron
> htmlhint app/renderer/*.html 


   Config loaded: /home/vipul/Documents/zulip-electron/.htmlhintrc

   Config loaded: /home/vipul/Documents/zulip-electron/.htmlhintrc

   Config loaded: /home/vipul/Documents/zulip-electron/.htmlhintrc

   Config loaded: /home/vipul/Documents/zulip-electron/.htmlhintrc

Scanned 4 files, no errors found (15 ms).
```

Output for Windows-
```
   Config loaded: C:\Users\kanis\Documents\GitHub\zulip-electron\.htmlhintrc

Scanned 4 files, no errors found (15 ms).
```

I want to ask if the extra config loaded lines are ok in ubuntu? Currently, there is an open issue [here](https://github.com/htmlhint/HTMLHint/issues/187) for this. One of the comments suggest to use single quotes, i.e. why i used them, in first case, but they don't seem to working well for Windows. Windows takes 'app/renderer/*.html' wholly as a file name, i guess.

vsvipul commented:
@akashnimare changing the npm script to -
```
"lint-html": "./node_modules/.bin/htmlhint \"app/renderer/*.html\" ",
```
fixes console output and linting for both Windows and Ubuntu. Tested on both. LGTM.

vsvipul commented:
@akashnimare a reminder for this.

akashnimare commented:
Thanks, @vsvipul.
